### ISSUE TITLE
Decreasing some dialogue depth and width

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5835]
There are many ways that I suggest the dialogues could be made more consistent and hence easier to use.
One simple change is that some dialogues are very deep.  This is not just annoying, but they may become unusable on some screen resolutions.  

There are two dialogues in particular.
a) **_Prepare > Date > Use Date_** is very deep and I think quite easy to change somewhat.
b) **File > Open from File** is similarly very deep.  There seems to be a gap at the top on the left of the dialogue.  On the right it is quite "full", but I suggest the preview at the top could be made smaller.  I also note that the dialogue could be made less wide.  There seems to be an unnecessary gap in the middle of the dialogue.

I think these are the two main "culprits" on depth.
On width (not so vital):
c) The Prepare > Dates > Make Date dialogue seems unnecessarily wide.
d) So does the Model > Fit Model> Fit Model Keyboard
I am sure there are more!

**Comment by shadrackkibet** [comment]
I think it will be good to discuss and agree on some of these things and document them so that it is one of the resources for developers. Good that @lilyclements will soon be involved again. She wrote the document [Making Dialogs.docx](https://github.com/africanmathsinitiative/R-Instat/files/4774199/Making.Dialogs.docx) attached which is still relevant however it might need updating.

**Comment by lloyddewit** [comment]
@shadrackkibet Thanks for sharing the document. 

@lilyclements This appears to be a sort of style guide that could be applied to any UI. If you have time at some point, please could you tell me a little about the document's background? 

@africanmathsinitiative/developers I'm interested in any R-Instat development documentation that may exist. I'd like to collect it together and share it on the Taiga Wiki, or in some other repository. I currently have:
- R-Instat Developer Documentation from @maxwellfundi 
- Adding Data Sets in R-Instat from @berylwaswa 

Please let me know if you're aware of any other developer documentation, thank you

**Comment by lilyclements** [comment]
@lloyddewit For the "Making dialogs" document, I wouldn't say I wrote it at all - I just took the appropriate bits out of a Windows guide and popped it into a document.
The document it is based off guidelines here - https://docs.microsoft.com/en-us/windows/win32/uxguide/guidelines
Perhaps that should be noted in the document.

@shadrackkibet you mentioned that some bits may need updating. This is most definitely a discussion for elsewhere, however, I am interested in what aspects need updating?
That guide was put together in 2017 so I'm sure the software has moved on. However, I also left in 2017 so I'm not as aware of design changes made since then.
